File: Praktikum Pemrograman I Modul 4/Praktikum I Modul4  07-11-2023/Python/PRAK403-2310817110015-AKHMAD_CHAIDAR_ANANDA.py
# input 
a, b = map(int, input().split())

# deklarasi variabel perulangan
# untuk memudahkan membaca variabel loop
i = a
j = b

# kondisi jika a kurang dari b
if (a < b):
    # nested loops
    for i in range(a, b):
        while j >= a:
            print(f"{i} {j} - ", end=' ')
            i += 1
            j -= 1
            # kondisi jika i bernilai b dan j bernilai a, maka loop akan berhenti 
            if (i == b and j == a):
                print(f"{i} {j}")
                i += 1
                j -= 1
                

elif (a > b):
    # nested loop
    while i >= b:
        for j in range(b, a):
            print(f"{i} {j} - ", end=' ')
            i -= 1
            j += 1
        # kondisi jika i bernilai b dan j bernilai a, maka loop akan berhenti 
        if (i == b and j == a):
            print(f"{i} {j}")
            i -= 1
            j += 1

File: Praktikum Pemrograman I Modul 4/Praktikum I Modul4  07-11-2023/C/PRAK403-2310817110015-AKHMAD_CHAIDAR_ANANDA.c
#include<stdio.h>

int main(){

    // deklarasi variabel
    int a, b;

    // input
    scanf("%d %d", &a, &b);

    // deklarasi variabel perulangan
    int i, j;

    if (a < b){
        for(i = a; i <= b; i++){
            for(j = b; j >= a; j--){
                if (j != a) {
                    printf("%d %d - ",i, j);
                    i += 1;
                } else {
                    printf("%d %d", i, j);
                }
            }
        }
    } else if (a > b) {
        for(i = a; i >= b; i--){
            for(j = b; j <= a; j++){
                if (j != a){
                    printf("%d %d - ",i, j);
                    i -= 1;
                } else {
                    printf("%d %d", i, j);
                }
            }
        }
    }
    
    return 0;
}